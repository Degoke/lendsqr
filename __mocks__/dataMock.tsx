import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
