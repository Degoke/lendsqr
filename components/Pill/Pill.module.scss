.pill {
  border-radius: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 90px;
  height: 30px;
  font-size: 14px;

  &.pending {
    background-color: rgba(233, 178, 0, 0.1);
    color: #e9b200;
  }

  &.blacklisted {
    background-color: rgba(228, 3, 59, 0.1);
    color: rgba(228, 3, 59, 1);
  }

  &.inactive {
    background-color: rgba(84, 95, 125, 0.06);
    color: rgba(84, 95, 125, 1);
  }

  &.active {
    background-color: rgba(57, 205, 98, 0.06);
    color: rgba(57, 205, 98, 1);
  }
}
