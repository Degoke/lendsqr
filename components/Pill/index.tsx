import { UserStatus } from '../../utils/types/user'
import styles from './Pill.module.scss'

type Props = {
  type: UserStatus
}

export default function Pill({ type }: Props) {
  return (
    <div className={`${styles.pill} ${styles[type]}`}>
      <p>{type}</p>
    </div>
  )
}
