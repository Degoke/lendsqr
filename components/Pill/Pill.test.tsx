import { render, screen } from '@testing-library/react'
import Pill from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'

describe('Pill tests', () => {
  it('renders Pill component unchanged', () => {
    const { container } = render(<Pill type="inactive" />)
    expect(container).toMatchSnapshot()
  })
})
