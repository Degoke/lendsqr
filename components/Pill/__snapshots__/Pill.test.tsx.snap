// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill tests renders Pill component unchanged 1`] = `
<div>
  <div
    class="pill inactive"
  >
    <p>
      inactive
    </p>
  </div>
</div>
`;
